dvc/remote/base.py

```diff
@@ -697,27 +697,45 @@ def checksum_to_path_info(self, checksum):
     def list_cache_paths(self, prefix=None, progress_callback=None):
         raise NotImplementedError
 
-    def all(self, *args, **kwargs):
-        # NOTE: The list might be way too big(e.g. 100M entries, md5 for each
-        # is 32 bytes, so ~3200Mb list) and we don't really need all of it at
-        # the same time, so it makes sense to use a generator to gradually
-        # iterate over it, without keeping all of it in memory.
-        for path in self.list_cache_paths(*args, **kwargs):
+    def cache_checksums(self, prefix=None, progress_callback=None):
+        """Iterate over remote cache checksums.
+
+        If `prefix` is specified, only checksums which begin with `prefix`
+        will be returned.
+        """
+        for path in self.list_cache_paths(prefix, progress_callback):
             try:
                 yield self.path_to_checksum(path)
             except ValueError:
                 logger.debug(
                     "'%s' doesn't look like a cache file, skipping", path
                 )
```

Comment:
This is makes push/pull/gc very verbose now to the point that I'm not sure it's worth logging?
```
...
2020-03-31 18:49:52,911 DEBUG: '/Users/pmrowla/git/remote-1m-files/.dvc/cache/c5/d84d5879b6266fe8624802e99ff211.dir.unpacked/000360c8.txt' doesn't look like a cache file, skipping
2020-03-31 18:49:52,935 DEBUG: '/Users/pmrowla/git/remote-1m-files/.dvc/cache/c5/d84d5879b6266fe8624802e99ff211.dir.unpacked/00001f77.txt' doesn't look like a cache file, skipping
2020-03-31 18:49:53,122 DEBUG: '/Users/pmrowla/git/remote-1m-files/.dvc/cache/c5/d84d5879b6266fe8624802e99ff211.dir.unpacked/000569a4.txt' doesn't look like a cache file, skipping
...
```

Prior to #3557 this exception handler was just a `pass` in `RemoteBASE.all()`

Reply:
@pmrowla It is verbose in `-v` verbose mode, but at least we know that it does skip it. Maybe it is time to support `-vv` and hide this, "assuming" and state logs under it. Will need to look into it, added a comment to https://github.com/iterative/dvc/issues/2329 .